compileopts/config.go

```diff
@@ -0,0 +1,128 @@
+// Package compileopts contains the configuration for a single to-be-built
+// binary.
+package compileopts
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/tinygo-org/tinygo/goenv"
+)
+
+// Config keeps all configuration affecting the build in a single struct.
+type Config struct {
+	Options        *Options
+	Target         *TargetSpec
```

Comment:
Is there a reason that there is a pointer here?

Reply:
You mean `*TargetSpec`? It's used as a pointer elsewhere, so using it as a pointer here as well.